lib/screens/selfReport/SelfReportList.dart

```diff
@@ -133,28 +133,46 @@ class _SelfReportListState extends State<SelfReportList> {
                     otherUID: widget.otherUID,
                     recurrenceReport: recurrenceReport));
               } else if (state is SelfReportRecurrenceReportSaved) {
-                Navigator.of(context).pop();
-                Navigator.of(context).pop();
-                Navigator.of(context).pop();
-                Navigator.of(context).pop();
-                Navigator.of(context).push(MaterialPageRoute(
-                    builder: (context) => SelfReportOption(
-                          location: widget.location,
-                          cityId: widget.cityId,
-                          isHealthStatusChanged: false,
-                        )));
-                Navigator.of(context).push(MaterialPageRoute(
-                    builder: (context) => SelfReportList(
-                          location: widget.location,
-                          cityId: widget.cityId,
-                          analytics: Analytics.tappedDailyReport,
-                          otherUID: widget.otherUID,
-                          otherRecurrenceReport:
-                              (int.parse(widget.otherRecurrenceReport ?? '0') +
-                                      1)
-                                  .toString(),
-                          isHealthStatusChanged: false,
-                        )));
+                if (widget.otherUID == null) {
+                  Navigator.of(context).pop();
+                  Navigator.of(context).pop();
+                  Navigator.of(context).pop();
+                  Navigator.of(context).pop();
+                  Navigator.of(context).push(MaterialPageRoute(
+                      builder: (context) => SelfReportOption(
+                            isQuarantined: true,
+                            location: widget.location,
+                            cityId: widget.cityId,
+                            isHealthStatusChanged: false,
+                          )));
+                  Navigator.of(context).push(MaterialPageRoute(
+                      builder: (context) => SelfReportList(
+                            location: widget.location,
+                            cityId: widget.cityId,
+                            analytics: Analytics.tappedDailyReport,
+                            otherUID: widget.otherUID,
+                            otherRecurrenceReport: (int.parse(
+                                        widget.otherRecurrenceReport ?? '0') +
+                                    1)
+                                .toString(),
+                            isHealthStatusChanged: false,
+                          )));
```

Comment:
Gaes, Ini ada ide yang bisa membuat simplekah, supaya tidak banyak navigator2 seperti ini? @jaluowen @gaganugraha33 @aldiiii

Reply:
Nah intinya sih buat ngereplace value `isHealthStatusChanged` jadi false yang nilai awalnya dikirim dari page lain, tadinya mau pasang bloc profile aja di page ini tapi kaya kurang tepat aja gitu request data ke firebase cuma buat 1 field aja. @anandarh @gaganugraha33 @aldiiii

Reply:
Line 137 - 140 itu untuk menutup apa? @jaluowen

Reply:
Setelah reset itu ada 2 pop up yang di close, terus sisanya ngeclose halaman SelfReportList sama SelfReportOption buat ngereplace value si `isHealthStatusChanged ` @anandarh

Reply:
Untuk sementara biar ga banyak nulis `Navigator.of(context).pop();`
Coba bisa ga kalo dibuat public global method untuk melakukan pop looping aja di bagian utilities? @jaluowen 
Misal:
`
void popUntil(BuildContext context, {@required int multiplication}) {
  if (multiplication != null) {
    for(int i = multiplication ; i >= 1; i-- ) {
      Navigator.of(context).pop();
    }
  }
}
`

Reply:
done mas bram @anandarh

Reply:
atau bisa gini 

`
int count = 0;
Navigator.of(context).popUntil((_) => count++ >= 2);
`

@jaluowen